pkg/controller/resourceclaim/controller.go

```diff
@@ -308,10 +308,11 @@ func (ec *Controller) podNeedsWork(pod *v1.Pod) (bool, string) {
 			return true, fmt.Sprintf("internal error while checking for claim: %v", err)
 		}
 
-		if checkOwner &&
-			resourceclaim.IsForPod(pod, claim) != nil {
-			// Cannot proceed with the pod unless that other claim gets deleted.
-			return false, "conflicting claim needs to be removed by user"
+		if checkOwner {
+			if err := resourceclaim.IsForPod(pod, claim); err != nil {
```

Comment:
https://github.com/kubernetes/kubernetes/blob/e07314860c7100e4bf68837f7a6a4f9d8e394c81/staging/src/k8s.io/dynamic-resource-allocation/resourceclaim/resourceclaim.go#L156-L168 is documented as "errors need no wrapping", so perhaps just return the `err.Error` here?

Reply:
done